### DISCUSSION TITLE
Is possible use the Firebase + Torus Authentication ?

### DISCUSSION THREAD

**Discussion by pr0mming** [Web3Auth/Web3Auth#823] [Help]
Hello there again,

I have implemented the Web3Auth + Firebase with this guide: https://web3auth.io/docs/guides/firebase

I got it, and works perfectly, I gonna share this code of the guide:

```
await web3Auth?.connectTo(WALLET_ADAPTERS.OPENLOGIN, {
	loginProvider: 'jwt',
	extraLoginOptions: {
		id_token: 'my access firebase token',
		verifierIdField: 'sub',
		domain: 'http://localhost:3000'
	}
})
```
I notice this code make the connection with Firebase and Web3Auth, but I need a verifier, so I need have an Adapter for configure the verifier:

```
const openloginAdapter = new OpenloginAdapter({
	adapterSettings: {
		clientId,
		network: environment?.isProduction ? 'mainnet' : 'testnet',
		uxMode: 'redirect',
		loginConfig: {
			jwt: {
				clientId,
				name: 'Blabla Web3',
				verifier: 'my verifier :)',
				typeOfLogin: 'jwt'
			}
		}
	}
})

web3auth.configureAdapter(openloginAdapter)
```

This code allow to specify the Web3Auth verifier, but I want to connect with Torus Wallet instead with Web3Auth, is this possible? Because in the Torus Wallet SignIn I can see the Social media methods, so I just suppose that is possible do the same workflow like OPENLOGIN but this time I want with Torus.

![image](https://user-images.githubusercontent.com/20020612/199997722-690fd963-1827-41e0-82b1-d96046139326.png)

So, I tried to do it by myself, first instead of use the OpenLoginAdapter I use the TorusWalletAdapter, the configuration is similar to the other, and I tried this: [](https://web3auth.io/docs/sdk/web/adapters/torus-evm)

 ```
const torusWalletAdapter = new TorusWalletAdapter({
	adapterSettings: {
		buttonPosition: "bottom-left",
	},
	loginSettings: {
		verifier: 'my verifier :)',
		login_hint: 'jwt'
	},
	initParams: {
		buildEnv: "testing",
	},
	chainConfig: {
		chainNamespace: CHAIN_NAMESPACES.EIP155,
		chainId: environment.web3Auth.chainConfig.chainId,
		rpcTarget: environment.web3Auth.chainConfig.rpcTarget,
		displayName: environment.web3Auth.chainConfig.displayName,
		blockExplorer: environment.web3Auth.chainConfig.blockExplorer,
		ticker: environment.web3Auth.chainConfig.ticker,
		tickerName: environment.web3Auth.chainConfig.tickerName
	},
	clientId
})

web3auth.configureAdapter(torusWalletAdapter);
```

And for the `connectTo` method I used the same code but this time instead of `OPENLOGIN` I use `TORUS_EVM`:

```
await web3Auth?.connectTo(WALLET_ADAPTERS.TORUS_EVM, {
	loginProvider: 'jwt',
	extraLoginOptions: {
		id_token: 'my access firebase token',
		verifierIdField: 'sub',
		domain: 'http://localhost:3000'
	}
})
```
But without luck I getting a verifier error:

![image](https://user-images.githubusercontent.com/20020612/199999140-82414ece-61bc-4e55-aed5-a8adaaab34d0.png)

Maybe my implementation is wrong, or is it impossible to do it ?

**Comment by rinzler99** [comment]
The problem might be in this code: 
await web3Auth?.connectTo(WALLET_ADAPTERS.TORUS_EVM, {
	loginProvider: 'jwt',
	extraLoginOptions: {
		id_token: 'my access firebase token',
		verifierIdField: 'sub',
		domain: 'http://localhost:3000'
	}
})


Try using openlogin instead of torus_evm it should work:
await web3Auth?.connectTo(WALLET_ADAPTERS.OPENLOGIN, {
	loginProvider: 'jwt',
	extraLoginOptions: {
		id_token: 'my access firebase token',
		verifierIdField: 'sub',
		domain: 'http://localhost:3000'
	}
})

**Reply by pr0mming** [reply]
Thanks for your reply, I tried this but without luck so far, but I tried keep the `TORUS_EVM` and the `verifier` with `google`, this works well ONLY with Google (Torus ask for confirm the logged account though), but I also want to login the user with Facebook, Apple, User + Password, etc. in my Firebase and stick the token to Torus Wallet

If I leave the field without value I can see the Torus Login screen:

```
const torusWalletAdapter = new TorusWalletAdapter({
   adapterSettings: {
   	buttonPosition: "bottom-left",
   },
   initParams: {
   	buildEnv: "testing",
   },
   chainConfig: {
   	chainNamespace: CHAIN_NAMESPACES.EIP155,
   	chainId: environment.web3Auth.chainConfig.chainId,
   	rpcTarget: environment.web3Auth.chainConfig.rpcTarget,
   	displayName: environment.web3Auth.chainConfig.displayName,
   	blockExplorer: environment.web3Auth.chainConfig.blockExplorer,
   	ticker: environment.web3Auth.chainConfig.ticker,
   	tickerName: environment.web3Auth.chainConfig.tickerName
   },
   clientId
})
```
Seems is not possible use the same Web3Auth verifier with `TorusWalletAdapter` :cry:

**Comment by rinzler99** [comment]
Hey My suggestion would be to use web3auth sdk, login using provided social logins or you can configure which logins do you want to show to user using whitelabelling.
If you want to use firebase with other logins you can try using aggregate verifier.
https://web3auth.io/docs/custom-authentication/verifiers#aggregate-verifiers
Then configure **torus wallet ui plugin** to use torus wallet. Refer below doc:
https://web3auth.io/docs/sdk/web/plugins/torus-wallet
